pulp_container/tests/functional/api/test_push_content.py

```diff
@@ -6,49 +6,145 @@
 
 from pulp_smash import cli, config, exceptions
 
-from pulp_container.tests.functional.utils import gen_container_client
+from pulp_container.tests.functional.constants import DOCKERHUB_PULP_FIXTURE_1
+from pulp_container.tests.functional.utils import del_user, gen_container_client, gen_user
 
-from pulpcore.client.pulp_container import DistributionsContainerApi
+from pulpcore.client.pulp_container import (
+    PulpContainerNamespacesApi,
+    RepositoriesContainerPushApi,
+)
 
 
-cfg = config.get_config()
+class PushRepoTestCase(unittest.TestCase):
+    """Verify whether images can be pushed to pulp."""
 
-api_client = gen_container_client()
-distributions_api = DistributionsContainerApi(api_client)
+    @classmethod
+    def setUpClass(cls):
+        """
+        Define APIs to use and pull images needed later in tests
+        """
+        api_client = gen_container_client()
+        cls.pushrepository_api = RepositoriesContainerPushApi(api_client)
+        cls.namespace_api = PulpContainerNamespacesApi(api_client)
 
-registry = cli.RegistryClient(cfg)
+        cfg = config.get_config()
+        cls.registry = cli.RegistryClient(cfg)
+        cls.registry.raise_if_unsupported(unittest.SkipTest, "Tests require podman/docker")
+        cls.registry_name = urlparse(cfg.get_base_url()).netloc
 
+        admin_user, admin_password = cfg.pulp_auth
+        cls.user_admin = {"username": admin_user, "password": admin_password}
+        cls.user_creator = gen_user(
+            [
+                "container.add_containerdistribution",
+                "container.view_containerdistribution",
+                "container.add_containernamespace",
+            ]
+        )
+        cls.user_reader = gen_user(["container.view_containerpushrepository"])
+        cls.user_helpless = gen_user([])
 
-class PushContentTestCase(unittest.TestCase):
-    """Verify whether images can be pushed to pulp."""
+        cls._pull("centos:7")
+        cls._pull(f"{DOCKERHUB_PULP_FIXTURE_1}:manifest_a")
+
+    @classmethod
+    def tearDownClass(cls):
+        """Delete api users."""
+        del_user(cls.user_creator)
+        del_user(cls.user_reader)
+        del_user(cls.user_helpless)
+
+    @classmethod
+    def _pull(cls, image_path, user=None):
+        """
+        Pull using specified user.
```

Comment:
can you please extend this doc block and add that we also ensuring we login with user if specified and logout after every pull